evm/utils/address.py

```diff
@@ -27,4 +27,8 @@ def private_key_to_address(private_key):
 
 
 def public_key_to_address(public_key):
-    return keccak(public_key[1:])[-20:]
+    if len(public_key) != 64:
+        raise ValueError(
```

Comment:
This strikes me as a good re-usable validation function since I bet we run into many more situations where we want to validate public keys.

I see this same code repeated in `evm/utils/secp256k1.py`.